notesapi/v1/views.py

```diff
@@ -275,20 +284,41 @@ def post(self, *args, **kwargs):  # pylint: disable=unused-argument
 
         Returns 400 request if bad payload is sent or it was empty object.
         """
-        if 'id' in self.request.data:
+        if not self.request.data or 'id' in self.request.data:
             return Response(status=status.HTTP_400_BAD_REQUEST)
 
+        total_notes = 0
         try:
+            total_notes = Note.objects.filter(
+                    user_id=self.request.data['user'], course_id=self.request.data['course_id']
+            ).count()
+            if total_notes >= settings.MAX_ANNOTATIONS_PER_COURSE:
+                raise AnnotationsLimitReachedError
+
             note = Note.create(self.request.data)
             note.full_clean()
         except ValidationError as error:
             log.debug(error, exc_info=True)
             return Response(status=status.HTTP_400_BAD_REQUEST)
+        except AnnotationsLimitReachedError:
+            error_message = ugettext_noop(
+                u'You can create up to {max_num_annotations_per_course} notes.'
+                u' You must remove some notes before you can add new ones.'
+            ).format(max_num_annotations_per_course=settings.MAX_ANNOTATIONS_PER_COURSE)
+            log.info(
+                u'Attempted to create more than %s annotations',
+                settings.MAX_ANNOTATIONS_PER_COURSE
+            )
+
+            # Gather metrics for New Relic so we can slice data in New Relic Insights
+            newrelic.agent.add_custom_parameter('notes.count', total_notes)
```

Comment:
Why we added this here. I think it should be near create_note method because we have to monitor the number of notes that user is creating most of the time.